### DISCUSSION TITLE
Detect discount by ID in the order (API and Webhook)

### DISCUSSION THREAD

**Discussion by its-anas** [Shopify/function-examples#221] [Feedback: Discounts APIs]
We can only identify the discount with its title but not an ID, and discounts can have the same title so we are unable to identify it.

**Comment by DuncanUszkay1** [comment]
Hey Anas 👋🏻 Not sure I follow- what API are you referring to and how does this relate to functions?

**Comment by its-anas** [comment]
Hi @DuncanUszkay1 
I was referring to Order API. In the discount applications part you get something like this:

"discount_applications": [
      {
        "type": "manual",
        "title": "custom discount",
        "description": "customer deserved it",
        "value": "2.0",
        "value_type": "fixed_amount",
        "allocation_method": "across",
        "target_selection": "explicit",
        "target_type": "line_item"
      },
      {
        "type": "script",
        "description": "my scripted discount",
        "value": "5.0",
        "value_type": "fixed_amount",
        "allocation_method": "across",
        "target_selection": "explicit",
        "target_type": "shipping_line"
      },
      {
        "type": "discount_code",
        "code": "SUMMERSALE",
        "value": "10.0",
        "value_type": "fixed_amount",
        "allocation_method": "across",
        "target_selection": "all",
        "target_type": "line_item"
      }

Assume you are creating an automatic discount using shopify functions, and after sale you want to check if a discount was applied to some line item in an order. There is no way to identify the discount with something other than the title (which is not unique i believe, because you can return any title on the function output), so getting the ID will help in identifying the discount and use it in query.

**Comment by thosakwe** [comment]
This would be a great addition, because it would allow app developers to bill based on # of discount applications, in a foolproof way.

It would also allow:
* Tagging orders where specific discounts applied
* Reporting total sales amounts when discounts were used
